docs/learn/protocols/iota2.0/core-concepts/communication-layer.md

```diff
@@ -118,17 +118,19 @@ The DRR scheduler iterates over all issuer queues in sequence and increments its
 
 ```vbnet
 IF Len(outbox) > 0
-    b = outbox[ID].head
-    WHILE ID.deficit < b.work_score OR b.timestamp > now() OR (parents(b) not scheduled OR not accepted)
-        ID.deficit += mana[ID]
+    FOR ID in round robin over all accounts
+        ID.deficit += ID.quantum
         IF ID.deficit > maxDeficit
             ID.deficit = maxDeficit
-        ID++
-    ID.deficit -= b.work_score
-    IF ID.deficit < 0
-        ID.deficit = 0
+        IF Len(outbox[ID]) == 0
+	        continue
+	    b = outbox[ID].head
+	    IF ID.deficit >= b.workScore AND (parents(b) scheduled OR accepted)
+	        break
```

Comment:
Remove this, let's just skip to the next iteration within the loop. If we want to be more precise, we can add a check if the outbox is empty to avoid iterating unnecessarily